src/Data/Acid/Archive.hs

```diff
@@ -39,9 +40,10 @@ entriesToListNoFail Fail{}            = []
 
 putEntry :: Entry -> Builder
 putEntry content
-    = putWord64le contentLength `mappend`
-      putWord16le contentHash `mappend`
-      fromLazyByteString content
+    = execPut $ do
+        putWord64le contentLength
+        putWord16le contentHash
+        putLazyByteString content
```

Comment:
Here it should also be possible to do the following without cereal at all. Let me know if you prefer that (and also the bytestring part above about the import). I'll update the pull request then.

    word64LE contentLength <>
    word16LE contentHash <>
    lazyByteString content